pizzeria-backend/Services/OrdersService.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Microsoft.EntityFrameworkCore;
+using pizzeria_backend.Models;
+using pizzeria_backend.Models.Interfaces;
+
+namespace pizzeria_backend.Services
+{
+    public interface IOrderService
+    {
+        public Task<Order> MakeOrder(OrderDto Order);
+        public Task<Order> GetOrder(int Id);
+        public Task<Order> DeleteOrder(int Id);
+        public Task<Order> ChangeOrderCompletion(int Id);
+        public Task<(List<int> productIds, List<int> addOnIds)> ValidateOrder(OrderDto Order);
+        public Task<List<Order>> GetOrders();
+
+    }
+    public class OrdersService : IOrderService
+    {
+        private readonly AppDbContext _context;
+
+        public OrdersService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Order> MakeOrder(OrderDto order)
+        {
+            var orderValidation = (await ValidateOrder(order));
+
+            if (orderValidation.productIds.Count != 0 && orderValidation.addOnIds.Count != 0)
+            {
+                return null;
```

Comment:
Another way to do this is throw a validation exception (or even a custom exception)

Reply:
Done